src/ray/raylet/scheduling/policy/scorer.cc

```diff
@@ -0,0 +1,108 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/raylet/scheduling/policy/scorer.h"
+
+#include <numeric>
+
+namespace ray {
+namespace raylet_scheduling_policy {
+
+double LeastResourceScorer::Score(const ResourceRequest &required_resources,
```

Comment:
can we have a bit more comments of the scoring algorithm? this is a fair large amount of code.

Reply:
also, is this just moving the code without changing any logic?

Reply:
> 

Yes, I just move these codes from `gcs_resoruce_scheduler` to this file and change nothing.